src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,7 +2450,21 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                    if (incompleteMember.ContainsDiagnostics)
+
+                    var containsErrorDiagnostic = false;
+                    foreach (var child in incompleteMember.EnumerateNodes())
```

Comment:
This feels like the wrong approach.  To my understanding the ContainsDiagnostics property is meant to check exactly what this loop is checking for.  Are we sure the bug isn't that ContainsDiagnostics is incorrectly reporting false here?  

Are there other examples in the parser where we have to Enumerate the child nodes to dig further into diagnostics?